src/rdkafka_partition.c

```diff
@@ -4421,6 +4421,91 @@ int rd_kafka_topic_partition_list_regex_cnt(
 }
 
 
+/**
+ * @brief Match function that returns true if topic is non regex.
+ */
+static RD_UNUSED int rd_kafka_topic_partition_not_regex(const void *elem,
+                                                        const void *opaque) {
+        const rd_kafka_topic_partition_t *rktpar = elem;
+        return *rktpar->topic != '^';
+}
+
+/**
+ * @brief Removes regexes from the list and returns the list without any regex
+ * in it.
+ */
+void rd_kafka_topic_partition_list_remove_regexes(
```

Comment:
should return the `rd_kafka_topic_partition_list_t *` and the `rktparlist` should be a single pointer and `const`